tests/objects/test_operation.py

```diff
@@ -339,12 +352,49 @@ def test_with_learning_parser(self, loop, contact_svc, data_svc, learning_svc, e
         fact = test_link.facts[0]
         assert fact.trait == 'host.ip.address'
         assert fact.value == '10.10.10.10'
+        knowledge_data = loop.run_until_complete(op_with_learning_parser.all_facts())
+        assert len(knowledge_data) == 1
+        assert knowledge_data[0].trait == 'host.ip.address'
+        assert knowledge_data[0].value == '10.10.10.10'
 
-    def test_without_learning_parser(self, loop, app_svc, contact_svc, data_svc, learning_svc, event_svc, op_without_learning_parser, make_test_link, make_test_result):
+    def test_without_learning_parser(self, loop, app_svc, contact_svc, data_svc, learning_svc, event_svc,
+                                     op_without_learning_parser, make_test_link, make_test_result):
         app_svc = app_svc(loop)  # contact_svc._save(...) needs app service registered
         test_link = make_test_link(5678)
         op_without_learning_parser.add_link(test_link)
         test_result = make_test_result(test_link.id)
         loop.run_until_complete(data_svc.store(op_without_learning_parser))
         loop.run_until_complete(contact_svc._save(test_result))
         assert len(test_link.facts) == 0
+
+    def test_facts(self, loop, app_svc, contact_svc, file_svc, data_svc, learning_svc, event_svc,
+                   op_with_learning_and_seeded, make_test_link, make_test_result, knowledge_svc, operation_agent):
+        test_link = make_test_link(9876)
+        op_with_learning_and_seeded.add_link(test_link)
+        # patch operation to make it 'realistic'
```

Comment:
Should the `op_with_learning_and_seeded` fixture be set up this way already so it doesn't need to be patched in tests that use it? (seems like this is the only one currently)

Reply:
It only needs this patching so that 'report' works properly, but I think you're right; it makes a lot of sense to put this all into the fixture. I'll adjust accordingly.